packages/block-library/src/button/index.js

```diff
@@ -36,6 +36,14 @@ export const settings = {
 		...a,
 		text: ( a.text || '' ) + text,
 	} ),
+	__experimentalLabel( attributes ) {
+		const customName = attributes?.metadata?.name;
+		if ( customName ) {
+			return customName;
+		}
+		const { text } = attributes;
+		return ! text || text.length === 0 ? __( 'Button' ) : text;
```

Comment:
I wonder if it's better to return undefined when the `text` is empty, so that block label hooks use a proper fallback based on the block or its variation title.

cc @t-hamano